ci/cron/src/Main.hs

```diff
@@ -34,42 +34,50 @@ import qualified Network.URI
 import qualified Options.Applicative as Opt
 import Safe (headMay)
 import qualified System.Directory as Directory
+import qualified System.Environment
 import qualified System.Exit as Exit
 import qualified System.IO.Extra as IO
 import qualified System.Process as System
 import qualified Text.Regex.TDFA as Regex
 
-shell_exit_code :: String -> IO (Exit.ExitCode, String, String)
-shell_exit_code cmd = do
-    System.readCreateProcessWithExitCode (System.shell cmd) ""
-
 die :: String -> Int -> String -> String -> IO a
 die cmd exit out err =
     fail $ unlines ["Subprocess:",
-                         cmd,
-                        "failed with exit code " <> show exit <> "; output:",
-                        "---",
-                        out,
-                        "---",
-                        "err:",
-                        "---",
-                        err,
-                        "---"]
+                    cmd,
+                    "failed with exit code " <> show exit <> "; output:",
+                    "---",
+                    out,
+                    "---",
+                    "err:",
+                    "---",
+                    err,
+                    "---"]
 
 shell :: String -> IO String
 shell cmd = System.readCreateProcess (System.shell cmd) ""
 
+proc :: [String] -> IO String
```

Comment:
Maybe pass it as `String -> [String]` instead to avoid `head` `tail`?

Reply:
That was my first attempt, mirroring what System.proc does, but at least in the context of this script I prefer building up a single list than having to treat the first element specially.

Is there a specific issue with calling `head` and `tail` here?

Reply:
Should be fine, I’m probably just a bit overzealous when it comes to avoiding partial functions. Happy to keep it.